stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1000,6 +1002,31 @@ internal constructor(
             Result.Success(chatSocket.disconnect())
         }
 
+    /**
+     * Fetches the current user.
+     * Works only if the user was previously set and the WS connections is closed.
+     */
+    public fun fetchCurrentUser(): Call<User> {
+        return CoroutineCall(userScope) {
+            logger.d { "[fetchCurrentUser] isUserSet: ${isUserSet()}, isSocketConnected: ${isSocketConnected()}" }
+            when {
+                !isUserSet() -> Result.Failure(Error.GenericError("User is not set, can't fetch current user"))
+                isSocketConnected() -> Result.Failure(
```

Comment:
If socket is connected, why we can't use this method?
Maybe we don't need to perform an API call to obtain it but return the `currentUser` property